views/js/runner/plugins/security/watermark.js

```diff
@@ -0,0 +1,441 @@
+                    if (!separatorsLength) {
+                        str += ' ';
+                    }
+                    return str;
+                });
+
+            /**
+             * Render a text long enough to cover the expected area
+             * @param {jQuery} $watermarkContent
+             */
+            this.renderText = ($watermarkContent, text) => {
+                for (let i = 0; i < 10; i++) {
+                    const actualHeight = $watermarkContent.get(0).offsetHeight;
+                    const expectedHeight = window.innerWidth + window.innerHeight; //enough to cover any degree of rotate
+                    if (actualHeight < expectedHeight) {
+                        const repeatedText = text.repeat(200);
+                        $watermarkContent.get(0).textContent += repeatedText;
+                    } else {
+                        break;
+                    }
+                }
+            };
+
+            /**
+             * Render text on a circular path
+             * @param {jQuery} $watermarkContent
+             */
+            this.renderCircle = ($watermarkContent, text) => {
+                $watermarkContent.html('');
+
+                const contentEl = $watermarkContent.get(0);
+                const boxSize = Math.trunc(Math.min(contentEl.offsetWidth, contentEl.offsetHeight));
+                const fontSizePropVal = getComputedStyle(contentEl).getPropertyValue('font-size');
+
+                const halfOfBoxSize = Math.trunc(boxSize / 2);
+                const fontSize = parseInt(fontSizePropVal) || 10;
+                const radius = halfOfBoxSize - fontSize;
+
+                const repeatedText = text.repeat(Math.trunc((3.15 * 2 * radius) / text.length / (fontSize / 4)));
+
+                $watermarkContent.html(
+                    watermarkCirclePathTpl({
+                        boxSize,
+                        repeatedText,
+                        s: halfOfBoxSize,
+                        f: fontSize,
+                        r: radius,
+                        sr: halfOfBoxSize + radius
+                    })
+                );
+            };
+
+            /**
+             * @param {string} val
+             * @returns {Promise<string>}
+             */
+            let validateUnlock = val =>
+                digest(val, this.pluginConfig.unlock.algorithm).then(hash => hash === this.pluginConfig.unlock.hash);
+
+            let doUnlock = () => {
+                isEnabled = !isEnabled;
+                if (this.isPluginEnabled()) {
+                    this.show();
+                } else {
+                    this.hide();
+                }
+            };
+
+            this.showUnlockDialog = () => {
+                this.removeUnlockListener();
+
+                const appendToEl = this.getAreaBroker().getContainer().find('.content-wrapper').get(0);
+                const dialogTpl = '<input type="password" autocomplete="off" class="tao-wmark-input" />';
```

Comment:
Could the dialog input have a text label? Or is it intentionally mysterious-looking?